janus_core/calculations/md.py

```diff
@@ -499,6 +509,80 @@ def __init__(
 
         self._parse_correlations()
 
+    def _set_time_step(self):
+        """Set time in fs and current dynamics step to info."""
+        time = (self.offset * self.timestep + self.dyn.get_time()) / units.fs
+        step = self.offset + self.dyn.nsteps
+        self.dyn.atoms.info["time_fs"] = time
+        self.dyn.atoms.info["step"] = step
+
+    def _prepare_restart(self) -> None:
+        """Prepare restart files, structure and offset."""
+        # Check offset can be read from steps
+        try:
+            with open(self.stats_file, encoding="utf8") as stats_file:
+                last_line = stats_file.readlines()[-1].split()
+            try:
+                self.offset = int(last_line[0])
+            except (IndexError, ValueError) as e:
+                raise ValueError("Unable to read restart file") from e
+        except FileNotFoundError as e:
+            raise FileNotFoundError("Unable to read restart file") from e
+
+        if self.restart_auto:
+            # Attempt to infer restart file
+            restart_stem = self._restart_stem
+
+            # Use restart_stem.name otherwise T300.0 etc. counts as extension
+            poss_restarts = restart_stem.parent.glob(f"{restart_stem.name}*.extxyz")
+            try:
+                last_restart = sorted(poss_restarts, key=getmtime)[-1]
+
+                # Read in last structure
+                self.struct = input_structs(
+                    struct_path=last_restart,
+                    read_kwargs=self.read_kwargs,
+                    sequence_allowed=False,
+                    arch=self.arch,
+                    device=self.device,
+                    model_path=self.model_path,
+                    calc_kwargs=self.calc_kwargs,
+                    set_calc=True,
+                    logger=self.logger,
+                )
+
+                # Infer last dynamics step
+                last_stem = last_restart.stem
+                try:
+                    # Remove restart_stem from filename
+                    # Use restart_stem.name otherwise T300.0 etc. counts as extension
+                    self.offset = int("".join(last_stem.split(f"{restart_stem.name}-")))
```

Comment:
The logic here is confusing. Would it be better to just split on `-` and take the appropriate index? Or is that not possible? Can you give me some examples of the filenames.

Reply:
Yeah it's not the nicest.

The default files are along the lines of `NaCl-nvt-T300.0-res-10.extxyz`. If you specify a file prefix: `prefix-res-10.extxyz`, or if you specify the file stem: `stem-10.extxyz`, so yes actually I think getting the last split with `-` should always work.